stories/DataTable/expandable/groupBy.stories.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import * as _ from 'lodash-es';
+
+import DataTable from '../../../src/DataTable/DataTable';
+import Row from '../../../src/DataTable/TableRow';
+import tableDataItems from '../../constants/sampleMovieData';
+
+const ExpandedComponent = ({ data, rowProps }) => {
+	let rowPropsPlus = {
+		...rowProps,
+		expandableRows: false,
+		columns: [
+			{
+				id: 0,
+				name: 'ID',
+				selector: row => row.id,
+				sortable: true,
+				wrap: true,
+			},
+			{
+				id: 1,
+				name: 'Title',
+				selector: row => row.title,
+				sortable: true,
+				wrap: true,
+			},
+			{
+				id: 2,
+				name: 'Year',
+				selector: row => row.year,
+				sortable: true,
+				wrap: true,
+			},
+			{
+				id: 3,
+				name: 'Runtime',
+				selector: row => row.runtime,
+				sortable: true,
+				wrap: true,
+			},
+		],
+	};
+	return _.map(data.groupData, item => {
+		return (
+			<div key={item.title}>
+				<Row {...rowPropsPlus} row={item} />
+			</div>
+		);
+	});
+};
+
+const columns = [
+	{
+		name: 'Group',
+		selector: row => row._groupKey,
+		sortable: true,
+		wrap: true,
+	},
+	{
+		name: 'Count',
+		selector: row => row.count,
+		sortable: true,
+		wrap: true,
+	},
+];
+
+const GroupByStory = props => {
+	const dataByGroupKey = _.groupBy(tableDataItems, row => props.groupKey(row));
+	const groupedData = _.map(dataByGroupKey, (group, key) => {
```

Comment:
I would move this logic inside the library when the user uses `groupBy` property.